docs/www/guide-wasm-elastic.md

```diff
@@ -0,0 +1,228 @@
+```
+
+expected config:
+
+```yaml
+   # Skips most of the checks performed at startup (i.e. memory, xfs)
+   # not recomended for production use
+   developer_mode: true
+   enable_coproc: true # enable wasm
+```
+
+## Create Wasm function
+In this example we use a folder for save all wasm functions, `/wasm_example`
+
+To create a wasm function we can use `rpk wasm` tools. It will generate a template giving us a simple build system and startup example.
+
+First, we need to set up the environment for defining and compacting
+wasm function, for it we can use `rpk wasm generate`
+
+```bash
+$ rpk wasm generate
+npm created project in /wasm_example/wasm
+$ /wasm_example tree wasm 
+wasm
+  ├── package.json
+  ├── src
+  │   └── wasm.js
+  ├── test
+  │   └── wasm.test.js
+  └── webpack.js
+```
+
+that command creates a folder with an example wasm
+definition in `/wasm_example/wasm`, now we just to go 
+`src/wasm.js`, and rename to `src/elastic.js`, remember
+the wasm file name is very important, from this we create
+the id for a wasm function. 
+
+```bash
+$ /wasm_example tree wasm/src
+wasm/src
+  └── elastic.js
+```
+
+Be sure to add elasticsearch to your install dependencies
+
+```bash
+$ npm install @elastic/elasticsearch
+```
+
+now we can edit `src/elastic.js` in order to apply our logic
+
+```js
+const {
+  SimpleTransform,
+  PolicyError,
+} = require("@vectorizedio/wasm-api");
+
+const {Client} = require('@elastic/elasticsearch')
+
+const client = new Client({ node: 'http://localhost:9200' })
+
+const transform = new SimpleTransform();
+/* Topics that fire the transform function */
+transform.subscribe(["produce"]);
+/* The strategy the transform engine will use when handling errors */
+transform.errorHandler(PolicyError.SkipOnFailure);
+/* function for validating record has the correct header key and value  */
+const containHeader = (key, value, record) =>
+  record.headers.some( header =>
+    header.headerKey.equals(Buffer.from(key)) &&
+    header.value.equals(Buffer.from(value))
+  )
+
+/* Transform function */
+transform.processRecord((recordBatch) => {
+  console.log("Applying wasm function")
+  return Promise.all(
+    recordBatch.records.map((record) => {
+      if (containHeader("logService", "elastic", record)) {
+        console.log("Index record value to Elastic Search")
+        return client.index({
+          index: "result_wasm",
+          type: "redpanda_wasm",
+          body: {
+            /*
+             index the record value into elastic search
+            */
+            record_message: record.value.toString()
+          }
+        })
+      }
+    })
+  ).then(() => {
+      /*
+       return a map with recordBatch indexed to elastic Search
+      */
+      const result = new Map();
+      result.set("elasticIndex", recordBatch)
+      /* 
+       Wasm processRecord function has to return a Map always,
+       although that map is empty map
+      */
+      return result
+  })
+    .catch((e) => {
```

Comment:
i don't think we should catch this exception. instead we should let it bubble up. and the script deregisters.

Reply:
Got it. we can remove it.